### DISCUSSION TITLE
Best practice for migrations when running Prisma 1 & 2  side-by-side

### DISCUSSION THREAD

**Discussion by patrickalbert** [prisma/prisma#16127] [Q&A] [answered]
My goal is to run Prisma 1 and 2 side by side for a time while doing a gradual migration of functionality. Based on the documentation this appears to be a supported use case.

So far I have successfully added Prisma 2 to my environment by following the migration documentation and running all queries which are indicated as non-breaking for backwards compatibility.

_**My current question is: What is the suggested process for updating the data structure when running both versions side-by-side?**_

Initially my thought would be to consider Prisma 2 the "master" schema and manage migrations through the new functionality, but I am concerned about inadvertently introducing breaking changes to the Prisma 1 implementation.

Any guidance / best practices would be greatly appreciated!

**Comment by pantharshit00** [comment, marked as answer]
Hello @patrickalbert 

I would actually use Prisma 1's migration system here for the time being as Prisma 2+'s introspection capabilities are far better so you can just introspect and carry on with that part of the application. 

After when you are also ready to switch migrations to P2+, you can just baseline(https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project#baseline-your-production-environment) and use the new migrate.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
